### PR TITLE
Add reproducible code for matrix exponential simulations

### PR DESCRIPTION
**Pull request by SamDuffield**
Reproducible code for the experiments in https://arxiv.org/abs/2311.12759

### CODE REVIEW

**Review by KaelanDt — COMMENTED** [review]
Looks good! I would also add the plots in the repo (what we did for other examples)

File: examples/matrix_exponentials/matrix_generation.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+def wishart(d: int, key: random.PRNGKey) -> jnp.ndarray:
+    n = 2 * d  # degrees of freedom
+    G = random.normal(key, shape=(d, n))
+    A_wishart = (G @ G.T) / n
+    return A_wishart
+
+
+def orthogonal(d: int, _) -> jnp.ndarray:
```
[the remark this answers is not in the document]
Comment (SamDuffield) [reply examples/matrix_exponentials/matrix_generation.py:12]: Unified signature with `wishart` but `ortho_group` is a scipy function that doesn't use jax random keys

File: examples/matrix_exponentials/matrix_generation.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from jax import random, numpy as jnp
+from scipy.stats import ortho_group
+
+
+def wishart(d: int, key: random.PRNGKey) -> jnp.ndarray:
+    n = 2 * d  # degrees of freedom
```
[the remark this answers is not in the document]
Comment (SamDuffield) [reply examples/matrix_exponentials/matrix_generation.py:6]: I think this one is ok haha have cleaned up run.py

File: examples/matrix_exponentials/run.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    return scan(
+        body_func,
+        jnp.zeros((samps.shape[1], samps.shape[1])),
+        jnp.arange(1, samps.shape[0]),
+    )[1]
+
+
+D = [64, 128, 256, 512]
+ERR_abs = np.zeros((args.n_repeats, len(D), NT))
```
[the remark this answers is not in the document]
Comment (SamDuffield) [reply examples/matrix_exponentials/run.py]: Changed `ERR` to `err`, I think the rest is ok

**Review by KaelanDt — APPROVED** [review]

**Pull request merged**